src/main/kotlin/org/jitsi/jicofo/conference/colibri/v2/ColibriV2SessionManager.kt

```diff
@@ -79,20 +79,27 @@ class ColibriV2SessionManager(
         clear()
     }
 
-    override fun removeParticipant(participant: Participant): Unit = synchronized(syncRoot) {
-        participant.setInviteRunnable(null)
+    override fun removeParticipant(participant: Participant) = removeParticipants(listOf(participant))
+    override fun removeParticipants(participants: Collection<Participant>) = synchronized(syncRoot) {
+        participants.forEach { it.setInviteRunnable(null) }
 
-        val participantInfo = participants[participant.endpointId]
-            ?: run {
-                logger.warn("No ParticipantInfo for participant $participant")
-                return
-            }
-        participantInfo.session.expire(participantInfo)
-        remove(participantInfo)
+        val participantInfos = participants.mapNotNull { this.participants[it.endpointId] }
+        if (participantInfos.size != participants.size) {
+            logger.error("Can not remove every participant requested, ParticipantInfo missing: " +
```

Comment:
Is this a "should not happen", a "could happen in a race", or normal?

Reply:
It's actually a "happens during normal operation because of awkward APIs": bridge goes down, JitsiMeetConference calls colibriSessionManager.removeBridges (which in v2 removes the participants), JitsiMeetConfernce calls removeParticipants. I've been thinking about how to avoid it, I think I'll just hide the message for now. When we remove colibri v1 we can clean it up.